### PR TITLE
Heaps algorithm iterative

### PR DESCRIPTION
**Pull request by grochedix**
### **Describe your change:**

Iterative Heap's algorithm. 

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: divide_and_conquer/heaps_algorithm_iterative.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    >>> heaps([0])
+    [(0,)]
+    >>> heaps([-1, 1])
+    [(-1, 1), (1, -1)]
+    >>> heaps([1, 2, 3])
+    [(1, 2, 3), (2, 1, 3), (3, 1, 2), (1, 3, 2), (2, 3, 1), (3, 2, 1)]
+    >>> from itertools import permutations
+    >>> sorted(heaps([1,2,3])) == sorted(permutations([1,2,3]))
+    True
+    """
```
Comment (cclauss) [inline divide_and_conquer/heaps_algorithm_iterative.py:25]:
```suggestion
    >>> all(sorted(heaps(x)) == sorted(permutations(x)) for x in ([], [0], [-1, 1], [1, 2, 3]))
    True
    """
```

Comment (cclauss) [reply divide_and_conquer/heaps_algorithm_iterative.py:25]: Then I think we are ready to land this one.

Comment (grochedix) [reply divide_and_conquer/heaps_algorithm_iterative.py:27]: Modified the iterative version as well as the recursive one ([#2475 ](https://github.com/TheAlgorithms/Python/pull/2475)) with your suggestion. 
Thanks for reviewing ! 👍

**Review by cclauss — APPROVED** [review]
LGTM

**Pull request merged**